src/BusInOut.cpp

```diff
@@ -0,0 +1,42 @@
+#include "BusInOut.h"
+
+BusInOut::BusInOut(int p0, int p1, int p2, int p3,
+                int p4, int p5, int p6, int p7,
+                int p8, int p9, int p10, int p11,
+                int p12, int p13, int p14, int p15) {
+    
+    int pins[] = {p0, p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12, p13, p14, p15};
+    for (int i=0; i<16; i++) {
+        _pin[i] = pins[i];
+    }
+}
+
+BusInOut::BusInOut(int  pins[]) {
+    for (int i=0; i<16; i++) {
+        _pin[i] = pins[i];
```

Comment:
Won't this throw out of bounds if pins[] isn't 16 ints long?

Reply:
Oh wait, I see. It's a 16 int array. However, if we have to put 16 ints in each time it's not very useful as then all logic should flow through BusInOut. It would be ideal if we could make this function variadic: https://en.cppreference.com/w/cpp/utility/variadic

Reply:
No, only because pins is declared length 16 in the header file. This does result in all the extra ints being 0 though, which works for mbed but won't work for Arduino.